shared/src/k8s/mod.rs

```diff
@@ -621,6 +621,69 @@ impl KubeInterface for KubeImpl {
     }
 }
 
+/// This deletes an Instance unless it has already been deleted by another node
+pub async fn try_delete_instance(
+    kube_interface: &impl KubeInterface,
+    instance_name: &str,
+    instance_namespace: &str,
+) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    match kube_interface
+        .delete_instance(instance_name, &instance_namespace)
```

Comment:
Should we surrounded this with a retry mechanism in case of failure?

Reply:
Good point. Fixed in bbe2fb815